src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -122,7 +123,18 @@ public boolean verifyUser(String username, String uniqueid) {
 
     @Override
     public boolean canUseLocalCache() {
-        boolean brokerSwitch = canSwitchToBroker();
+        boolean brokerSwitch;
+        try
+        {
+            brokerSwitch = canSwitchToBroker();
+        }
+        catch (DeveloperAuthenticationException e){
```

Comment:
i mean 
final DeveloperAuthenticationException e
I think it's a good practice to declare variable as final if you know it's not going to be changed.